packages/client/tests/functional/_utils/setupQueryEngine.ts

```diff
@@ -1,46 +1,22 @@
 import { enginesVersion } from '@prisma/engines'
 import { download } from '@prisma/fetch-engine'
-import type { Platform } from '@prisma/get-platform'
-import { getNodeAPIName } from '@prisma/get-platform'
-import { ClientEngineType } from '@prisma/internals'
-import fs from 'fs-extra'
 import path from 'path'
-import lockfile from 'proper-lockfile'
 
 /**
  * Ensures the correct Query Engine (`node-api`/`binary`) is present. This is required as
  * normally the downloading of the required engine is done in `getGenerators`. As the test
  * clients bypass this we need to ensure the correct engine is present.
- * @param clientEngineType
- * @param platform
  */
-export async function setupQueryEngine(clientEngineType: ClientEngineType, platform: Platform) {
+export async function setupQueryEngine() {
   const engineDownloadDir = path.join(__dirname, '..', '..', '..')
 
-  // Multiple jest workers can run this function at the same time
-  // we need to ensure that no 2 downloads run in parallel. In order to achieve
-  // that, we lock engine directory and do not unlock it again until download is done
-  const releaseLock = await lockfile.lock(engineDownloadDir, {
-    lockfilePath: path.join(engineDownloadDir, 'dir.lock'),
-    retries: {
-      forever: true,
+  await download({
```

Comment:
We download both at the start of the test setup in jest, so once only.

Reply:
Once per worker, isn't it? If so, we still need a lock so multiple workers won't start downloading at the same time

Reply:
Are you sure it is per worker? https://jestjs.io/docs/configuration#globalsetup-string

> This option allows the use of a custom global setup module, which must export a function (it can be sync or async). The function will be triggered once before all test suites and it will receive two arguments

Reply:
The setup phase of the query engine is now happening in the `_utils/globalSetup.js` file

Reply:
Hmm, this paragraph also hints at this likely being run in the main process but not in workers:

> Any global variables that are defined through globalSetup can only be read in globalTeardown. You cannot retrieve globals defined here in your test suites.

Reply:
Yea, we figured it out in private chat: `globalSetup` part is correct, but `await` for global set here in `beforeAll` is not. That await confused me, so I assumed it runs for every worker.

Reply:
So yes it is triggered once, but globals cannot be exported from there as Serhii pointed out, which means `await (global as any).queryEngineSetup` is actually undefined :see_no_evil: . So I will need to make the global setup async instead.

Reply:
Looks good now, i think dependency on `proper-lockfile' and `@types/proper-lockfile` can be removed now.

Reply:
Done.